### PR TITLE
Pttb 303 refactor delete user from admin controller

### PR DESCRIPTION
**Pull request by piyushmhrzn**
1. Single action controller "DeleteUserAction" used.
2. "Try-catch" functionality implemented. 
3. "Loggers" implemented from Exceptions.
4. "UserModel" & "UserService" dependency injected for action class.
5. "Doc Blocks" implemented.
6. "HttpResponse" trait implemented for uniform messages.
7. "Strict type" for id used.
8. "Enum constant" used to check role.

### CODE REVIEW

File: app/Enum/UserRoleEnum.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    // Human Resource
+    const HR = 'hr';
+
+    // Developer
+    const DEV = 'dev';
+
+    // Management
+    const MANAGEMENT = 'management';
+
+ }
```
Comment (sahajmalla) [inline app/Enum/UserRoleEnum.php:22]: add one space below line 22

File: app/Services/UserService.php [hunk 2]
```diff
@@ -165,4 +165,22 @@ public static function checkUserIdExists($id)
...
+    /**
+     *
+     * @param integer $id
+     * @return boolean
+     */
+    public function deleteUser(int $id): bool
+    {
+        $user = $this->userModel->where('id', $id)->firstOrFail();
+        
+        foreach($user->roles->toArray() as $role){
```
Comment (sahajmalla) [inline app/Services/UserService.php:178]: extract this foreach in a function.

**Review by sahajmalla — CHANGES_REQUESTED** [review]

File: app/Services/UserService.php (hunk 2 above)
Comment (piyushmhrzn) [reply app/Services/UserService.php:188]: created separate function for checking if user has role admin.

File: app/Enum/UserRoleEnum.php (hunk 1 above)
Comment (piyushmhrzn) [reply app/Enum/UserRoleEnum.php]: space added

File: app/Enum/UserRoleEnum.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
```
Comment (sahajmalla) [inline app/Enum/UserRoleEnum.php:23]: add one space below. Github is showing error

File: app/Services/UserService.php [hunk 4]
```diff
@@ -165,4 +165,32 @@ public static function checkUserIdExists($id)
...
+
+    /**
+     *
+     * @param object $user
+     * @return boolean
+     */
+    public function hasRoleAdmin(object $user): bool
+    {
+        foreach($user->roles->toArray() as $role){
+            if($role['role'] === UserRoleEnum::ADMIN){
```
Comment (sahajmalla) [inline app/Services/UserService.php:189]: use ARR::get

**Review by sahajmalla — CHANGES_REQUESTED** [review]

File: app/Enum/UserRoleEnum.php (hunk 3 above)
Comment (piyushmhrzn) [reply app/Enum/UserRoleEnum.php:23]: line 24 added

File: app/Services/UserService.php (hunk 4 above)
Comment (piyushmhrzn) [reply app/Services/UserService.php:189]: Used Arr::get($role, 'role') instead of $role['role']

**Review by sahajmalla — APPROVED** [review]

**Pull request merged**